WireAnalytics/Sources/WireAnalytics/Service/AnalyticsService.swift

```diff
@@ -42,31 +44,27 @@ public final class AnalyticsService: AnalyticsServiceProtocol {
     ///   - config: A config containing the authentication key and server host.
     ///   - logger: A function that logs a message on behalf of the service.
 
-    public convenience init(
-        config: Config?,
-        logger: @escaping (String) -> Void // TODO: [WPB-11881] import WireLogging and remove this workaround
-    ) {
+    public convenience init(config: Config?, deviceModel: String, deviceOS: String) {
         self.init(
             config: config,
             baseSegmentation: [
-                .deviceModel(UIDevice.current.model),
-                .deviceOS(UIDevice.current.systemVersion)
+                .deviceModel(deviceModel),
+                .deviceOS(deviceOS)
```

Comment:
`UIDevice` can only be access from the `MainActor` and this leads to either an error or many more code changes in Swift 6.